interface/src/error.rs

```diff
@@ -1,5 +1,5 @@
 use num_traits::{FromPrimitive, ToPrimitive};
-use serde::Serialize;
+use serde_derive::Serialize;
```

Comment:
You can still use this derive macro with the `serde` dependency but just including the `derive` feature, but up to you!